## Two Sample Models

We now compare independent Poissons with different rates (means).

In [1]:
import sys
sys.path.append("../../")


from BDA.stats import *
from BDA.plots import *

now importing: 
1. numpy as np 
2. from scipy import stats


now importing: 
3. matplotlib.pyplot as plt



In [2]:
import pandas as pd

pima = pd.read_csv("../data/Pima.csv", index_col = 0)
pima.head()

npreg  glu    bp  skin   bmi    ped  age type
1      5   86  68.0  28.0  30.2  0.364   24   No
2      7  195  70.0  33.0  25.1  0.163   55  Yes
3      5   77  82.0  41.0  35.8  0.156   35   No
4      0  165  76.0  43.0  47.9  0.259   26   No
5      0  107  60.0  25.0  26.4  0.133   23   No

In [6]:
## Calculate rates for each `type`
pima.groupby(by = "type")['npreg'].agg(['sum', 'count', 'mean'])

sum  count      mean
type                       
No    1282    417  3.074341
Yes   1011    215  4.702326

In [9]:
3.074/4.702

0.6537643555933645

For this analysis we will use noninformative prior Gamma(0.0001, 0.0001) and let the data elicit the information we need.

Thus, we obtain the following posteriors

$$
\theta | y_{No} \sim Gamma(1282.0001, 417.0001) \\
\text{ and } \\
\theta | y_{Yes} \sim Gamma(1011.0001, 215.0001)
$$

with posterior means $\theta_{no} = 3.074$ and $\theta_{yes} = 4.702$. We can obtain credible intervals for these estimates by simply taking 2.5th and 97.5th quantiles.

The posterior means have ratio $\theta_{No}/\theta_{Yes} = 0.654$ which is much lower than 1, alluding to a difference in rates.



In [13]:
qgamma(p=[0.025, 0.975], rate=417.0001, shape=1282.0001)

array([ 2.90833368,  3.24488883])

In [14]:
qgamma(p=[0.025, 0.975], rate=215.0001, shape=1011.0001)

array([ 4.41689623,  4.99656158])

We can now see that the 95% CIs do no overlap and can conclude that rates are significantly different from each other.